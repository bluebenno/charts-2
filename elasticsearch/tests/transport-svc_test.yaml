suite: test transport service
templates:
  - transport-svc.yaml
tests:
  - it: has default values
    asserts:
      - hasDocuments: { count: 1 }
      - isAPIVersion: { of: v1 }
      - isKind: { of: Service }
      - matchSnapshot: { path: spec }
      - equal:
          path: spec.type
          value: ClusterIP
      - contains:
          path: spec.ports
          content:
            port: 80
            targetPort: transport
            protocol: TCP
            name: transport

  - it: can set service type
    set:
      service.type: LoadBalancer
    asserts:
      - equal:
          path: spec.type
          value: LoadBalancer

  - it: can set service port
    set:
      service.port: 19300
    asserts:
      - equal:
          path: spec.ports[0].port
          value: 19300

  - it: can select pods
    asserts:
      - equal:
          path: spec.selector
          value:
            app.kubernetes.io/name: elasticsearch
            app.kubernetes.io/instance: RELEASE-NAME
