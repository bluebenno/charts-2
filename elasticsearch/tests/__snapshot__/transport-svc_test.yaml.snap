has default values:
  1: |
    ports:
    - name: transport
      port: 80
      protocol: TCP
      targetPort: transport
    selector:
      app.kubernetes.io/instance: RELEASE-NAME
      app.kubernetes.io/name: elasticsearch
    type: ClusterIP
